test/integration/template_insights_test.go

```diff
@@ -0,0 +1,240 @@
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+
+	"github.com/nuodb/nuodb-helm-charts/test/testlib"
+)
+
+func checkSidecarContainers(t *testing.T, containers []v1.Container, options *helm.Options, chartPath string) {
+	assert.NotEmpty(t, containers)
+	found := 0
+
+	for _, container := range containers {
+		t.Logf("Inspecting container %s in chart %s", container.Name, chartPath)
+		if container.Name == "insights" {
+			found++
+			assert.Equal(t, container.Image, fmt.Sprintf("%s/%s:%s",
+				options.SetValues["insights.image.registry"],
+				options.SetValues["insights.image.repository"],
+				options.SetValues["insights.image.tag"]))
+		} else if container.Name == "insights-config" {
+			found++
+			assert.Equal(t, container.Image, fmt.Sprintf("%s/%s:%s",
+				options.SetValues["insights.watcher.registry"],
+				options.SetValues["insights.watcher.repository"],
+				options.SetValues["insights.watcher.tag"]))
+			assert.Contains(t, container.Env, v1.EnvVar{
+				Name:  "FOLDER",
+				Value: "/etc/telegraf/telegraf.d/dynamic/",
+			})
+			assert.Contains(t, container.Env, v1.EnvVar{
+				Name:  "REQ_URL",
+				Value: "http://127.0.0.1:5000/reload",
+			})
+			if chartPath == testlib.ADMIN_HELM_CHART_PATH {
+				assert.Contains(t, container.Env, v1.EnvVar{
+					Name:  "LABEL",
+					Value: "nuodb.com/nuocollector-plugin in (release-name-nuodb-cluster0-admin, insights)",
+				})
+			} else {
+				assert.Contains(t, container.Env, v1.EnvVar{
+					Name:  "LABEL",
+					Value: "nuodb.com/nuocollector-plugin in (release-name-nuodb-cluster0-demo-database, insights)",
+				})
+			}
+		} else {
+			// This is probably the main container
+			continue
+		}
+		assert.Contains(t, container.VolumeMounts, v1.VolumeMount{
+			Name:      "insights-config",
+			MountPath: "/etc/telegraf/telegraf.d/dynamic/",
+		})
+		assert.Contains(t, container.VolumeMounts, v1.VolumeMount{
+			Name:      "log-volume",
+			MountPath: "/var/log/nuodb",
+		})
+	}
+
+	expectedContainersCount := 0
+	if options.SetValues["insights.enabled"] == "true" {
+		expectedContainersCount = 2
+	}
+	assert.Equal(t, expectedContainersCount, found)
+}
+
+func checkSpecVolumes(t *testing.T, volumes []v1.Volume, options *helm.Options, chartPath string) {
+	assert.NotEmpty(t, volumes)
+	found := false
+	for _, volume := range volumes {
+		if volume.Name == "insights-config" {
+			found = true
+			// Check that empty dir is mounted
+			assert.NotNil(t, volume.EmptyDir)
+		}
+	}
+	if options.SetValues["insights.enabled"] == "true" {
+		assert.True(t, found)
```

Comment:
assert.True tends to have a fairly incomprehensible failure message. @NikTJ777 always advocates for explicit failure messages. So something like `assert.True(t, found, "insights-config volume not found") might be helpful

Reply:
Thanks for the comment! I agree with you and added messages.